example for failure:
```
make e2e-test
mkdir -p /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache
podman run --rm \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup:/go/src/github.com/kiagnose/kubevirt-realtime-checkup:Z \
	-v /home/ralavi/go/src/github.com/kiagnose/kubevirt-realtime-checkup/_go-cache:/root/.cache/go-build:Z \
	-v /home/ralavi/.kube/sno03-cnvqe2-rdu2:/root/.kube:Z,ro \
	--workdir /go/src/github.com/kiagnose/kubevirt-realtime-checkup \
	-e KUBECONFIG=/root/.kube/kubeconfig \
	-e TEST_NAMESPACE=realtime-checkup-1 \
	-e TEST_CHECKUP_IMAGE=quay.io/ramlavi/kubevirt-realtime-checkup:devel \
	-e VM_UNDER_TEST_CONTAINER_DISK_IMAGE=quay.io/kiagnose/kubevirt-realtime-checkup-vm:main \
	docker.io/library/golang:1.19.4-bullseye \
	go test -v ./tests/...
=== RUN   TestTests
Running Suite: Tests Suite - /go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests
========================================================================================
Random Seed: 1701606425

Will run 1 of 1 specs
------------------------------
• [FAILED] [13.413 seconds]
Checkup execution [It] should complete successfully
/go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/checkup_test.go:80

  [FAILED] checkup failed: {
  	"spec.param.oslatDuration": "10m",
  	"spec.param.oslatLatencyThresholdMicroSeconds": "45",
  	"spec.param.vmUnderTestContainerDiskImage": "quay.io/kiagnose/kubevirt-realtime-checkup-vm:main",
  	"spec.param.vmUnderTestTargetNodeName": "",
  	"spec.timeout": "15m",
  	"status.completionTimestamp": "2023-12-03T12:27:13Z",
  	"status.failureReason": "admission webhook \"virtualmachineinstances-create-validator.kubevirt.io\" denied the request: spec.domain.memory.hugepages must be requested when NUMA topology strategy is set in spec.domain.cpu.numa.guestMappingPassthrough",
  	"status.startTimestamp": "2023-12-03T12:27:13Z",
  	"status.succeeded": "false"
  }
  In [It] at: /go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/checkup_test.go:90 @ 12/03/23 12:27:17.556
------------------------------

Summarizing 1 Failure:
  [FAIL] Checkup execution [It] should complete successfully
  /go/src/github.com/kiagnose/kubevirt-realtime-checkup/tests/checkup_test.go:90

Ran 1 of 1 Specs in 13.414 seconds
FAIL! -- 0 Passed | 1 Failed | 0 Pending | 0 Skipped
--- FAIL: TestTests (13.42s)
FAIL
FAIL	github.com/kiagnose/kubevirt-realtime-checkup/tests	13.423s
FAIL
make: *** [Makefile:67: e2e-test] Error 1
```

--- a/tests/checkup_test.go
+++ b/tests/checkup_test.go
@@ -21,6 +21,7 @@ package tests
 
 import (
 	"context"
+	"encoding/json"
 	"fmt"
 	"time"
 
@@ -77,7 +78,21 @@ var _ = Describe("Checkup execution", func() {
 	})
 
 	It("should complete successfully", func() {
-		Eventually(getJobConditions, 15*time.Minute, 5*time.Second).Should(
+		Eventually(func() []batchv1.JobCondition {
+			jobConditions, err := getJobConditions()
+			Expect(err).NotTo(HaveOccurred())
+
+			for _, jobCondition := range jobConditions {
+				if jobCondition.Type == batchv1.JobFailed && jobCondition.Status == corev1.ConditionTrue {
+					configMap, err := client.CoreV1().ConfigMaps(testNamespace).Get(context.Background(), testConfigMapName, metav1.GetOptions{})
+					Expect(err).NotTo(HaveOccurred())
+
+					Fail(fmt.Sprintf("checkup failed: %+v", prettifyData(configMap.Data)))
+				}
+			}
+
+			return jobConditions
+		}, 15*time.Minute, 5*time.Second).Should(
 			ContainElement(MatchFields(IgnoreExtras, Fields{
 				"Type":   Equal(batchv1.JobComplete),
 				"Status": Equal(corev1.ConditionTrue),
@@ -91,6 +106,12 @@ var _ = Describe("Checkup execution", func() {
 		Expect(configMap.Data["status.failureReason"]).To(BeEmpty(), fmt.Sprintf("should be empty %+v", configMap.Data))
 	})
 })
+
+func prettifyData(data map[string]string) string {
+	dataPrettyJSON, err := json.MarshalIndent(data, "", "\t")
+	Expect(err).NotTo(HaveOccurred())
+	return string(dataPrettyJSON)
+}
 
 func setupCheckupPermissions() {
 	var (
@@ -321,11 +342,11 @@ func pointer[T any](v T) *T {
 	return &v
 }
 
-func getJobConditions() []batchv1.JobCondition {
+func getJobConditions() ([]batchv1.JobCondition, error) {
 	checkupJob, err := client.BatchV1().Jobs(testNamespace).Get(context.Background(), testCheckupJobName, metav1.GetOptions{})
 	if err != nil {
-		return []batchv1.JobCondition{}
+		return nil, err
 	}
 
-	return checkupJob.Status.Conditions
+	return checkupJob.Status.Conditions, nil
 }

